### PR TITLE
[14.0][FIX] shopfloor_single_product_transfer: check package

### PR DESCRIPTION
**Pull request by mt-software-de**
check scanned package

### CODE REVIEW

**Comment by mt-software-de** [comment]
cc @jbaudoux @mmequignon

**Review by jbaudoux — APPROVED** [review]
LGTM

**Comment by jbaudoux** [comment]
@mmequignon Can you drop a review?

Ref: rau-235

File: shopfloor_single_product_transfer/services/single_product_transfer.py [hunk 1]
```diff
@@ -718,7 +733,7 @@ def _set_quantity__by_package(self, move_line, package, confirmation=False):
             location = package.location_id
             handlers = self._set_quantity__by_location_handlers()
             response = self._use_handlers(
-                handlers, move_line, location, confirmation=confirmation
+                handlers, move_line, location, package, confirmation=confirmation
```
Comment (simahawk) [inline shopfloor_single_product_transfer/services/single_product_transfer.py]:
```suggestion
                handlers, move_line, location, package=package, confirmation=confirmation
```

File: shopfloor_single_product_transfer/services/single_product_transfer.py [hunk 2]
```diff
@@ -882,10 +897,10 @@ def set_location(self, selected_line_id, package_id, barcode):
         }
         search = self._actions_for("search")
         search_result = search.find(barcode, types=handlers_by_type.keys())
+        package = self.env["stock.quant.package"].browse(package_id)
         handler = handlers_by_type.get(search_result.type)
         if handler:
-            return handler(move_line, search_result.record)
-        package = self.env["stock.quant.package"].browse(package_id)
+            return handler(move_line, search_result.record, package)
```
Comment (simahawk) [inline shopfloor_single_product_transfer/services/single_product_transfer.py]:
```suggestion
            return handler(move_line, search_result.record, package=package)
```

File: shopfloor_single_product_transfer_force_package/services/single_product_transfer.py [hunk 3]
```diff
@@ -8,15 +9,20 @@
...
+            and not package
             and not move_line.result_package_id
         ):
             message = self.msg_store.location_requires_package()
             return self._response_for_set_quantity(
                 move_line, message=message, asking_confirmation=False
             )
-        return super()._set_quantity__check_location(move_line, location, confirmation)
+        return super()._set_quantity__check_location(
+            move_line, location, package, confirmation
```
Comment (simahawk) [inline shopfloor_single_product_transfer_force_package/services/single_product_transfer.py]:
```suggestion
            move_line, location, package=package, confirmation=confirmation
```

File: shopfloor_single_product_transfer_unique_order_at_location/services/single_product_transfer.py [hunk 4]
```diff
@@ -10,9 +11,11 @@ class ShopfloorSingleProductTransfer(Component):
     _inherit = "shopfloor.single.product.transfer"
 
     def _set_quantity__check_location(
-        self, move_line, location=False, confirmation=False
+        self, move_line, location=False, package=False, confirmation=False
     ):
-        res = super()._set_quantity__check_location(move_line, location, confirmation)
+        res = super()._set_quantity__check_location(
+            move_line, location, package, confirmation
```
Comment (simahawk) [inline shopfloor_single_product_transfer_unique_order_at_location/services/single_product_transfer.py]:
```suggestion
            move_line, location, package=package, confirmation=confirmation
```

**Review by simahawk — APPROVED** [review]
LG

**Review by mmequignon — COMMENTED** [review]
LG, however instead of changing the signature for both `set_quantity` and `set_destination`, I would've created a dedicated handler.
Would be easier to modify the behavior in one enpoint only.

**Comment by mt-software-de** [comment]
@simahawk could you merge this one?

**Comment by mt-software-de** [comment]
> @simahawk could you merge this one?

or maybe @jbaudoux ?

**Review by TDu — APPROVED** [review]

**Comment by mt-software-de** [comment]
ping @jbaudoux  could you retry to merge this?

**Pull request merged**